Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -70,14 +75,122 @@ HTMLAttachmentElement::~HTMLAttachmentElement() = default;
 
 Ref<HTMLAttachmentElement> HTMLAttachmentElement::create(const QualifiedName& tagName, Document& document)
 {
-    return adoptRef(*new HTMLAttachmentElement(tagName, document));
+    Ref<HTMLAttachmentElement> attachment = adoptRef(*new HTMLAttachmentElement(tagName, document));
+    if (DeprecatedGlobalSettings::attachmentWideLayoutEnabled() && DeprecatedGlobalSettings::attachmentElementEnabled()) {
+        ASSERT(attachment->m_implementation == Implementation::Legacy);
+        ASSERT(!attachment->renderer()); // Switch to modern style *must* be done before renderer is created!
+        attachment->m_implementation = Implementation::Modern;
+        if (auto shadow = attachment->userAgentShadowRoot())
+            attachment->ensureModernShadowTree(*shadow);
+        else
+            attachment->ensureUserAgentShadowRoot();
+    }
+    return attachment;
+}
+
+void HTMLAttachmentElement::didAddUserAgentShadowRoot(ShadowRoot& root)
+{
+    if (m_implementation == Implementation::Modern)
+        ensureModernShadowTree(root);
+}
+
+void HTMLAttachmentElement::ensureModernShadowTree(ShadowRoot& root)
+{
+    ASSERT(m_implementation == Implementation::Modern);
+    if (m_elementWithTitle)
+        return;
+
+    constexpr auto styleSheet = R"CSS(
+attachment::-webkit-attachment-container {
+    display: grid;
+    max-width: 500px;
+    background-color: lightgray;
+    border-radius: 8px;
+    font-family: "Lucida Grande";
```

Comment:
You probably want a system font value: https://developer.mozilla.org/en-US/docs/Web/CSS/font#system_font_values

Although if used in the relevant contexts, the correct font should just inherit down.

Reply:
It will be used inside an editor with user-selected fonts around, so I think I need to specify our choice of font here.
Thanks for the tip about system fonts.